dkg/exchanger.go

```diff
@@ -65,6 +66,10 @@ func (stb *dataByPubkey) get(sigType sigType) (map[core.PubKey][]core.ParSignedD
 		return nil, ok
 	}
 
+	if len(data) != stb.numVals {
```

Comment:
I thought the data is only set once and then it includes all the validators? I don't understand why this check is required or why this works?